menas/src/main/scala/za/co/absa/enceladus/menas/WebSecurityConfig.scala

```diff
@@ -137,76 +67,82 @@ class WebSecurityConfig {
-        .ignoringAntMatchers("/api/login")
+          .ignoringAntMatchers("/api/login")
         .and()
-        .exceptionHandling()
-        .authenticationEntryPoint(restAuthenticationEntyPoint)
+          .exceptionHandling()
+          .authenticationEntryPoint(restAuthenticationEntyPoint)
         .and()
-        .authorizeRequests()
-        .antMatchers("/index.html", "/resources/**", "/generic/**",
-          "/service/**", "/webjars/**", "/css/**", "/components/**")
-        .permitAll()
-        .anyRequest().authenticated()
+          .authorizeRequests()
+          .antMatchers("/index.html", "/resources/**", "/generic/**",
+            "/service/**", "/webjars/**", "/css/**", "/components/**")
+          .permitAll()
+        .anyRequest()
+          .authenticated()
         .and()
-        .formLogin()
-        .loginProcessingUrl("/api/login")
-        .successHandler(authenticationSuccessHandler)
-        .failureHandler(authenticationFailureHandler)
-        .permitAll()
+          .formLogin()
+          .loginProcessingUrl("/api/login")
+          .successHandler(authenticationSuccessHandler)
+          .failureHandler(authenticationFailureHandler)
+          .permitAll()
         .and()
         .logout()
-        .logoutUrl("/api/logout")
-        .logoutSuccessHandler(logoutSuccessHandler)
-        .permitAll()
-        .clearAuthentication(true)
-        .deleteCookies("JSESSIONID")
-        .invalidateHttpSession(true)
+          .logoutUrl("/api/logout")
+          .logoutSuccessHandler(logoutSuccessHandler)
+          .permitAll()
+          .clearAuthentication(true)
+          .deleteCookies("JSESSIONID")
+          .invalidateHttpSession(true)
     }
 
+
     override def configure(auth: AuthenticationManagerBuilder) {
-      auth
-        .inMemoryAuthentication()
-          .passwordEncoder(NoOpPasswordEncoder.getInstance())
-          .withUser(username)
-            .password(password)
-            .authorities("ROLE_USER")
+      this.getMenasAuthentication().configure(auth)
     }
 
-//    override def configure(auth: AuthenticationManagerBuilder) {
-//      auth.authenticationProvider(activeDirectoryLdapAuthenticationProvider())
-//        .authenticationProvider(kerberosServiceAuthenticationProvider())
-//    }
-//
-//    @Bean
-//    override def authenticationManagerBean() = {
-//      super.authenticationManagerBean()
-//    }
+    private def getMenasAuthentication(): MenasAuthentication = {
+      authMechanism.toLowerCase match {
+        case "inmemory" => {
+          logger.info("Using InMemory Menas authentication")
+          beanFactory.getBean(classOf[InMemoryMenasAuthentication])
+        }
+        case "kerberos" => {
+          logger.info("Using Kerberos Menas Authentication")
+          beanFactory.getBean(classOf[KerberosMenasAuthentication])
+        }
+        case _ => throw new IllegalArgumentException(s"Invalid authentication mechanism - use one of: inmemory, kerberos")
```

Comment:
Why not also have this on the new line as previous matches.

And brackets are redundant here.